### PR TITLE
Enable big file translation (for massive back translation)

### PR DESCRIPTION
**Pull request by vince62s**
without running out of memory.

This PR uses the same mechanism as in preprocess.py
The use case is "Massive back translation"
When you need to translate millions of segments, you will run out of memory.
This PR will cut in shard_size chunks and translate sequentially.

### CODE REVIEW

**Comment by vince62s** [comment]
@bpopeters if you can have a quick look.

File: onmt/opts.py [hunk 1]
```diff
@@ -511,6 +511,14 @@ def translate_opts(parser):
               help='Source directory for image or audio files')
     group.add('--tgt', '-tgt',
                        help='True target sequence (optional)')
+    group.add('--shard_size', '-shard_size', type=int, default=1000000,
```
Comment (bpopeters) [inline onmt/opts.py]: 1 million seems like a very large default shard size. Is not sharding at translation time a better default? 

I just noticed that setting `shard_size=0` does not have the intended effect -- it fails to read the file at all. I'll try to have a fix for this soon.

File: translate.py [hunk 2]
```diff
@@ -3,22 +3,39 @@
...
-
+import codecs
+from itertools import islice
 from onmt.utils.logging import init_logger
 from onmt.translate.translator import build_translator
 
 import onmt.opts as opts
 
 
+def split_corpus(path, shard_size):
```
Comment (bpopeters) [inline translate.py]: Is there a way to do this without copy-pasting from preprocess.py?

File: onmt/opts.py (hunk 1 above)
Comment (vince62s) [reply onmt/opts.py]: it all depends on the machine you're doing this :) but yes I'll set it to 10k
Yes I also noticed the shard_size=0 not being effective, letting you fix it.

File: translate.py (hunk 2 above)
Comment (vince62s) [reply translate.py]: yes, I'll move this to utils/misc.py

**Pull request merged**